INFO 2023-09-20 16:07:56,895 autoreload 6748 11436 Watching for file changes with StatReloader
INFO 2023-09-20 16:10:04,118 autoreload 6748 11436 D:\DjangoProject\homework_project\homework_project\settings.py changed, reloading.
INFO 2023-09-20 16:10:04,550 autoreload 8068 4136 Watching for file changes with StatReloader
INFO 2023-09-20 16:10:12,214 basehttp 8068 13836 "GET / HTTP/1.1" 200 1416
INFO 2023-09-20 16:18:28,579 autoreload 8692 6260 Watching for file changes with StatReloader
INFO 2023-09-20 16:18:31,706 basehttp 8692 12024 "GET / HTTP/1.1" 200 2016
WARNING 2023-09-20 16:18:36,567 log 8692 12024 Not Found: /index/
WARNING 2023-09-20 16:18:36,568 basehttp 8692 12024 "GET /index/ HTTP/1.1" 404 2433
INFO 2023-09-20 16:19:18,131 autoreload 8692 6260 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:19:18,556 autoreload 11436 13592 Watching for file changes with StatReloader
WARNING 2023-09-20 16:19:26,834 log 11436 14712 Not Found: /index/
WARNING 2023-09-20 16:19:26,834 basehttp 11436 14712 "GET /index/ HTTP/1.1" 404 2433
INFO 2023-09-20 16:20:40,795 autoreload 11436 13592 D:\DjangoProject\homework_project\hw1_app\urls.py changed, reloading.
INFO 2023-09-20 16:20:41,233 autoreload 12628 14436 Watching for file changes with StatReloader
WARNING 2023-09-20 16:20:47,572 log 12628 11248 Not Found: /
WARNING 2023-09-20 16:20:47,573 basehttp 12628 11248 "GET / HTTP/1.1" 404 2403
INFO 2023-09-20 16:21:03,784 basehttp 12628 11248 "GET /index HTTP/1.1" 301 0
INFO 2023-09-20 16:21:03,788 basehttp 12628 14756 "GET /index/ HTTP/1.1" 200 2011
INFO 2023-09-20 16:22:20,867 autoreload 12628 14436 D:\DjangoProject\homework_project\hw1_app\urls.py changed, reloading.
INFO 2023-09-20 16:22:21,305 autoreload 13592 4324 Watching for file changes with StatReloader
INFO 2023-09-20 16:22:31,734 basehttp 13592 8912 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:22:36,296 basehttp 13592 8912 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:22:40,463 basehttp 13592 8912 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:24:39,735 autoreload 13592 4324 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:24:40,265 autoreload 12100 14584 Watching for file changes with StatReloader
INFO 2023-09-20 16:26:11,615 autoreload 12100 14584 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:26:12,080 autoreload 5932 14680 Watching for file changes with StatReloader
INFO 2023-09-20 16:29:57,179 autoreload 5932 14680 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:29:57,608 autoreload 12676 5860 Watching for file changes with StatReloader
INFO 2023-09-20 16:30:07,979 basehttp 12676 8576 "GET /about/ HTTP/1.1" 200 2203
WARNING 2023-09-20 16:30:08,005 log 12676 8576 Not Found: /about/me.jpg
WARNING 2023-09-20 16:30:08,006 basehttp 12676 8576 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:31:19,872 autoreload 12676 5860 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:31:20,302 autoreload 8628 11336 Watching for file changes with StatReloader
INFO 2023-09-20 16:31:24,980 basehttp 8628 1764 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:31:30,705 basehttp 8628 1764 "GET /about/ HTTP/1.1" 200 2211
WARNING 2023-09-20 16:31:30,735 log 8628 1764 Not Found: /about/hw1_app/me.jpg
WARNING 2023-09-20 16:31:30,735 basehttp 8628 1764 "GET /about/hw1_app/me.jpg HTTP/1.1" 404 2475
INFO 2023-09-20 16:32:56,860 autoreload 8628 11336 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:32:57,302 autoreload 14908 5396 Watching for file changes with StatReloader
INFO 2023-09-20 16:33:01,673 basehttp 14908 6836 "GET /about/ HTTP/1.1" 200 2228
WARNING 2023-09-20 16:33:01,695 log 14908 6836 Not Found: /about/homework_project/hw1_app/me.jpg
WARNING 2023-09-20 16:33:01,696 basehttp 14908 6836 "GET /about/homework_project/hw1_app/me.jpg HTTP/1.1" 404 2526
INFO 2023-09-20 16:33:02,605 basehttp 14908 6836 "GET /about/ HTTP/1.1" 200 2228
WARNING 2023-09-20 16:33:02,619 log 14908 6836 Not Found: /about/homework_project/hw1_app/me.jpg
WARNING 2023-09-20 16:33:02,620 basehttp 14908 6836 "GET /about/homework_project/hw1_app/me.jpg HTTP/1.1" 404 2526
INFO 2023-09-20 16:33:24,359 autoreload 14908 5396 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:33:24,838 autoreload 5680 1860 Watching for file changes with StatReloader
INFO 2023-09-20 16:33:29,384 basehttp 5680 12100 "GET /about/ HTTP/1.1" 200 2229
WARNING 2023-09-20 16:33:29,406 log 5680 12100 Not Found: /homework_project/hw1_app/me.jpg
WARNING 2023-09-20 16:33:29,406 basehttp 5680 12100 "GET /homework_project/hw1_app/me.jpg HTTP/1.1" 404 2508
INFO 2023-09-20 16:33:31,715 basehttp 5680 12100 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:33:34,009 basehttp 5680 12100 "GET /about/ HTTP/1.1" 200 2229
WARNING 2023-09-20 16:33:34,028 log 5680 12100 Not Found: /homework_project/hw1_app/me.jpg
WARNING 2023-09-20 16:33:34,029 basehttp 5680 12100 "GET /homework_project/hw1_app/me.jpg HTTP/1.1" 404 2508
INFO 2023-09-20 16:35:01,421 autoreload 5680 1860 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:35:01,858 autoreload 8692 2148 Watching for file changes with StatReloader
INFO 2023-09-20 16:35:08,212 basehttp 8692 6136 "GET /about/ HTTP/1.1" 200 2245
INFO 2023-09-20 16:35:09,950 basehttp 8692 6136 "GET /about/ HTTP/1.1" 200 2245
INFO 2023-09-20 16:35:12,522 basehttp 8692 6136 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:35:14,210 basehttp 8692 6136 "GET /about/ HTTP/1.1" 200 2245
INFO 2023-09-20 16:36:51,809 autoreload 8692 2148 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:36:52,244 autoreload 11388 14240 Watching for file changes with StatReloader
INFO 2023-09-20 16:36:56,453 basehttp 11388 14736 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:36:56,471 log 11388 14736 Not Found: /about/me.jpg
WARNING 2023-09-20 16:36:56,471 basehttp 11388 14736 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:36:57,712 basehttp 11388 14736 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:36:57,728 log 11388 14736 Not Found: /about/me.jpg
WARNING 2023-09-20 16:36:57,728 basehttp 11388 14736 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:36:58,518 basehttp 11388 14736 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:36:58,532 log 11388 14736 Not Found: /about/me.jpg
WARNING 2023-09-20 16:36:58,532 basehttp 11388 14736 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:36:59,239 basehttp 11388 14736 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:36:59,253 log 11388 14736 Not Found: /about/me.jpg
WARNING 2023-09-20 16:36:59,254 basehttp 11388 14736 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:37:12,948 basehttp 11388 14736 "GET / HTTP/1.1" 200 2010
INFO 2023-09-20 16:38:24,731 autoreload 11388 14240 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:38:25,180 autoreload 14796 6120 Watching for file changes with StatReloader
INFO 2023-09-20 16:38:42,717 basehttp 14796 15316 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 16:38:46,294 basehttp 14796 15316 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:38:46,393 log 14796 15316 Not Found: /about/me.jpg
WARNING 2023-09-20 16:38:46,394 basehttp 14796 15316 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:41:08,198 autoreload 14796 6120 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:41:08,693 autoreload 13600 15340 Watching for file changes with StatReloader
INFO 2023-09-20 16:41:11,409 basehttp 13600 14896 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:41:11,433 log 13600 14896 Not Found: /about/me.jpg
WARNING 2023-09-20 16:41:11,434 basehttp 13600 14896 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:41:12,193 basehttp 13600 14896 "GET /about/ HTTP/1.1" 200 2202
WARNING 2023-09-20 16:41:12,208 log 13600 14896 Not Found: /about/me.jpg
WARNING 2023-09-20 16:41:12,209 basehttp 13600 14896 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:42:04,412 autoreload 13600 15340 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:42:04,858 autoreload 1436 12780 Watching for file changes with StatReloader
INFO 2023-09-20 16:42:05,888 basehttp 1436 3460 "GET /about/ HTTP/1.1" 200 2442
WARNING 2023-09-20 16:42:05,909 log 1436 3460 Not Found: /about/me.jpg
WARNING 2023-09-20 16:42:05,909 basehttp 1436 3460 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:42:06,938 basehttp 1436 3460 "GET /about/ HTTP/1.1" 200 2442
WARNING 2023-09-20 16:42:06,953 log 1436 3460 Not Found: /about/me.jpg
WARNING 2023-09-20 16:42:06,954 basehttp 1436 3460 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:42:07,473 basehttp 1436 3460 "GET /about/ HTTP/1.1" 200 2442
WARNING 2023-09-20 16:42:07,489 log 1436 3460 Not Found: /about/me.jpg
WARNING 2023-09-20 16:42:07,490 basehttp 1436 3460 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:42:31,792 autoreload 1436 12780 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:42:32,235 autoreload 14756 15312 Watching for file changes with StatReloader
INFO 2023-09-20 16:42:34,871 basehttp 14756 9408 "GET /about/ HTTP/1.1" 200 2444
WARNING 2023-09-20 16:42:34,888 log 14756 9408 Not Found: /about/me.jpg
WARNING 2023-09-20 16:42:34,889 basehttp 14756 9408 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:42:35,493 basehttp 14756 9408 "GET /about/ HTTP/1.1" 200 2444
WARNING 2023-09-20 16:42:35,508 log 14756 9408 Not Found: /about/me.jpg
WARNING 2023-09-20 16:42:35,508 basehttp 14756 9408 "GET /about/me.jpg HTTP/1.1" 404 2451
INFO 2023-09-20 16:43:51,475 autoreload 14756 15312 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:43:51,944 autoreload 15140 10024 Watching for file changes with StatReloader
INFO 2023-09-20 16:45:29,387 autoreload 15140 10024 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:45:29,824 autoreload 916 11152 Watching for file changes with StatReloader
INFO 2023-09-20 16:45:33,386 basehttp 916 11344 "GET /about/ HTTP/1.1" 200 2488
INFO 2023-09-20 16:45:35,055 basehttp 916 11344 "GET /about/ HTTP/1.1" 200 2488
INFO 2023-09-20 16:45:35,812 basehttp 916 11344 "GET /about/ HTTP/1.1" 200 2488
INFO 2023-09-20 16:45:37,305 basehttp 916 11344 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 16:45:38,472 basehttp 916 11344 "GET /about/ HTTP/1.1" 200 2488
INFO 2023-09-20 16:49:11,107 autoreload 916 11152 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:49:11,544 autoreload 11500 14396 Watching for file changes with StatReloader
INFO 2023-09-20 16:49:16,413 basehttp 11500 10004 "GET /about/ HTTP/1.1" 200 2449
WARNING 2023-09-20 16:49:16,437 log 11500 10004 Not Found: /img/me.jpg
WARNING 2023-09-20 16:49:16,437 basehttp 11500 10004 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:49:18,868 basehttp 11500 10004 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 16:49:20,576 basehttp 11500 10004 "GET /about/ HTTP/1.1" 200 2449
WARNING 2023-09-20 16:49:20,672 log 11500 10004 Not Found: /img/me.jpg
WARNING 2023-09-20 16:49:20,673 basehttp 11500 10004 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:49:58,109 autoreload 11500 14396 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:49:58,541 autoreload 15032 15068 Watching for file changes with StatReloader
INFO 2023-09-20 16:50:05,965 basehttp 15032 8696 "GET /about/ HTTP/1.1" 200 2452
WARNING 2023-09-20 16:50:05,988 log 15032 8696 Not Found: /img/me.jpg
WARNING 2023-09-20 16:50:05,989 basehttp 15032 8696 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:50:07,071 basehttp 15032 8696 "GET /about/ HTTP/1.1" 200 2452
WARNING 2023-09-20 16:50:07,087 log 15032 8696 Not Found: /img/me.jpg
WARNING 2023-09-20 16:50:07,088 basehttp 15032 8696 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:50:07,741 basehttp 15032 8696 "GET /about/ HTTP/1.1" 200 2452
WARNING 2023-09-20 16:50:07,757 log 15032 8696 Not Found: /img/me.jpg
WARNING 2023-09-20 16:50:07,757 basehttp 15032 8696 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:50:50,139 autoreload 15032 15068 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:50:50,575 autoreload 5104 10936 Watching for file changes with StatReloader
INFO 2023-09-20 16:50:55,252 basehttp 5104 14696 "GET /about/ HTTP/1.1" 200 2447
WARNING 2023-09-20 16:50:55,274 log 5104 14696 Not Found: /img/me.jpg
WARNING 2023-09-20 16:50:55,275 basehttp 5104 14696 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:50:55,983 basehttp 5104 14696 "GET /about/ HTTP/1.1" 200 2447
WARNING 2023-09-20 16:50:55,997 log 5104 14696 Not Found: /img/me.jpg
WARNING 2023-09-20 16:50:55,998 basehttp 5104 14696 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:53:01,873 autoreload 5104 10936 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 16:53:02,332 autoreload 14204 10616 Watching for file changes with StatReloader
INFO 2023-09-20 16:54:18,747 autoreload 14968 13380 Watching for file changes with StatReloader
INFO 2023-09-20 16:54:30,837 basehttp 14968 12844 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 16:54:33,642 basehttp 14968 12844 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 16:54:33,733 log 14968 12844 Not Found: /img/me.jpg
WARNING 2023-09-20 16:54:33,734 basehttp 14968 12844 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 16:59:04,384 basehttp 14968 12844 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 16:59:04,405 log 14968 12844 Not Found: /img/me.jpg
WARNING 2023-09-20 16:59:04,405 basehttp 14968 12844 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 17:06:12,723 basehttp 14968 14844 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 17:06:12,742 log 14968 14844 Not Found: /img/me.jpg
WARNING 2023-09-20 17:06:12,743 basehttp 14968 14844 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 17:08:25,310 basehttp 14968 14844 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 17:08:27,443 basehttp 14968 14844 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 17:08:27,535 log 14968 14844 Not Found: /img/me.jpg
WARNING 2023-09-20 17:08:27,535 basehttp 14968 14844 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 17:10:04,502 basehttp 14968 14844 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 17:10:06,133 basehttp 14968 14844 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 17:10:06,152 log 14968 14844 Not Found: /img/me.jpg
WARNING 2023-09-20 17:10:06,153 basehttp 14968 14844 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 17:18:20,189 basehttp 14968 14252 "GET /about/ HTTP/1.1" 200 2489
WARNING 2023-09-20 17:18:20,205 log 14968 14252 Not Found: /img/me.jpg
WARNING 2023-09-20 17:18:20,206 basehttp 14968 14252 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 17:24:34,786 autoreload 4308 2860 Watching for file changes with StatReloader
INFO 2023-09-20 17:24:44,974 basehttp 4308 1080 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 17:24:46,378 basehttp 4308 1080 "GET /about/ HTTP/1.1" 200 2507
WARNING 2023-09-20 17:24:46,482 log 4308 1080 Not Found: /about/img/me.jpg
WARNING 2023-09-20 17:24:46,483 basehttp 4308 1080 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-20 17:24:50,007 basehttp 4308 1080 "GET /about/ HTTP/1.1" 200 2507
WARNING 2023-09-20 17:24:50,026 log 4308 1080 Not Found: /about/img/me.jpg
WARNING 2023-09-20 17:24:50,027 basehttp 4308 1080 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-20 17:27:12,704 autoreload 8348 12024 Watching for file changes with StatReloader
ERROR 2023-09-20 17:27:19,859 log 8348 6756 Internal Server Error: /
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: index() takes 0 positional arguments but 1 was given
ERROR 2023-09-20 17:27:19,861 basehttp 8348 6756 "GET / HTTP/1.1" 500 61585
INFO 2023-09-20 17:27:37,732 autoreload 8348 12024 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 17:27:38,186 autoreload 15920 13380 Watching for file changes with StatReloader
INFO 2023-09-20 17:27:43,130 basehttp 15920 12968 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 17:28:29,415 basehttp 15920 12968 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 17:29:45,408 autoreload 15920 13380 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 17:29:45,848 autoreload 15148 13976 Watching for file changes with StatReloader
INFO 2023-09-20 21:59:30,593 autoreload 6844 14384 Watching for file changes with StatReloader
INFO 2023-09-20 21:59:34,781 basehttp 6844 13640 "GET / HTTP/1.1" 200 2144
WARNING 2023-09-20 21:59:34,920 log 6844 13640 Not Found: /favicon.ico
WARNING 2023-09-20 21:59:34,921 basehttp 6844 13640 "GET /favicon.ico HTTP/1.1" 404 2448
INFO 2023-09-20 21:59:43,155 basehttp 6844 4532 "GET /about/ HTTP/1.1" 200 2505
WARNING 2023-09-20 21:59:43,259 log 6844 4532 Not Found: /img/me.jpg
WARNING 2023-09-20 21:59:43,259 basehttp 6844 4532 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:00:00,046 basehttp 6844 4532 "GET / HTTP/1.1" 200 2144
INFO 2023-09-20 22:04:59,232 autoreload 6844 14384 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:04:59,674 autoreload 10436 2316 Watching for file changes with StatReloader
INFO 2023-09-20 22:05:08,718 basehttp 10436 16216 "GET / HTTP/1.1" 200 2134
INFO 2023-09-20 22:05:21,514 basehttp 10436 16216 "GET /about/ HTTP/1.1" 200 2502
WARNING 2023-09-20 22:05:21,535 log 10436 16216 Not Found: /img/me.jpg
WARNING 2023-09-20 22:05:21,536 basehttp 10436 16216 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:05:30,370 basehttp 10436 16216 "GET / HTTP/1.1" 200 2134
INFO 2023-09-20 22:05:48,041 basehttp 10436 16216 "GET /about/ HTTP/1.1" 200 2502
WARNING 2023-09-20 22:05:48,059 log 10436 16216 Not Found: /img/me.jpg
WARNING 2023-09-20 22:05:48,060 basehttp 10436 16216 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:06:23,722 basehttp 10436 16216 "GET /about/ HTTP/1.1" 200 2502
WARNING 2023-09-20 22:06:23,744 log 10436 16216 Not Found: /img/me.jpg
WARNING 2023-09-20 22:06:23,745 basehttp 10436 16216 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:10:11,268 basehttp 10436 16216 "GET / HTTP/1.1" 200 2134
INFO 2023-09-20 22:14:18,260 autoreload 10436 2316 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:14:18,701 autoreload 3656 15064 Watching for file changes with StatReloader
INFO 2023-09-20 22:16:46,238 autoreload 3656 15064 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:16:46,683 autoreload 13660 6100 Watching for file changes with StatReloader
INFO 2023-09-20 22:16:58,433 basehttp 13660 11452 "GET / HTTP/1.1" 200 2030
INFO 2023-09-20 22:17:37,278 autoreload 13660 6100 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:17:37,714 autoreload 3848 6896 Watching for file changes with StatReloader
INFO 2023-09-20 22:17:47,711 basehttp 3848 7764 "GET / HTTP/1.1" 200 2030
INFO 2023-09-20 22:18:04,340 basehttp 3848 7764 "GET /about/ HTTP/1.1" 200 2374
WARNING 2023-09-20 22:18:04,365 log 3848 7764 Not Found: /img/me.jpg
WARNING 2023-09-20 22:18:04,366 basehttp 3848 7764 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:18:12,470 basehttp 3848 7764 "GET / HTTP/1.1" 200 2030
INFO 2023-09-20 22:20:41,164 autoreload 3848 6896 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:20:41,613 autoreload 14156 14076 Watching for file changes with StatReloader
INFO 2023-09-20 22:20:48,982 basehttp 14156 10236 "GET / HTTP/1.1" 200 2131
INFO 2023-09-20 22:21:01,875 basehttp 14156 10236 "GET /about/ HTTP/1.1" 200 2374
WARNING 2023-09-20 22:21:01,900 log 14156 10236 Not Found: /img/me.jpg
WARNING 2023-09-20 22:21:01,901 basehttp 14156 10236 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:25:31,016 basehttp 14156 10236 "GET /about/ HTTP/1.1" 200 2374
WARNING 2023-09-20 22:25:31,032 log 14156 10236 Not Found: /img/me.jpg
WARNING 2023-09-20 22:25:31,033 basehttp 14156 10236 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:25:32,985 basehttp 14156 10236 "GET /about/ HTTP/1.1" 200 2374
WARNING 2023-09-20 22:25:33,000 log 14156 10236 Not Found: /img/me.jpg
WARNING 2023-09-20 22:25:33,001 basehttp 14156 10236 "GET /img/me.jpg HTTP/1.1" 404 2445
INFO 2023-09-20 22:30:30,805 autoreload 14156 14076 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-20 22:30:31,256 autoreload 7428 6628 Watching for file changes with StatReloader
INFO 2023-09-20 22:30:38,308 basehttp 7428 15508 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-09-20 22:30:38,336 log 7428 15508 Not Found: /about/img/me.jpg
WARNING 2023-09-20 22:30:38,337 basehttp 7428 15508 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-20 22:30:40,324 basehttp 7428 15508 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-09-20 22:30:40,340 log 7428 15508 Not Found: /about/img/me.jpg
WARNING 2023-09-20 22:30:40,341 basehttp 7428 15508 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-21 09:08:32,478 basehttp 7428 11956 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-09-21 09:08:32,501 log 7428 11956 Not Found: /about/img/me.jpg
WARNING 2023-09-21 09:08:32,501 basehttp 7428 11956 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-21 09:08:33,325 basehttp 7428 11956 "GET / HTTP/1.1" 200 2131
INFO 2023-09-21 09:20:25,071 autoreload 7428 6628 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:20:25,519 autoreload 7480 9460 Watching for file changes with StatReloader
INFO 2023-09-21 09:20:38,596 basehttp 7480 1972 "GET /about/ HTTP/1.1" 200 2491
INFO 2023-09-21 09:22:10,320 autoreload 7480 9460 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:22:10,759 autoreload 11360 9672 Watching for file changes with StatReloader
INFO 2023-09-21 09:22:23,209 basehttp 11360 15716 "GET /about/ HTTP/1.1" 200 2489
INFO 2023-09-21 09:22:26,269 basehttp 11360 15716 "GET /about/ HTTP/1.1" 200 2489
INFO 2023-09-21 09:22:52,165 autoreload 11360 9672 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:22:52,606 autoreload 15416 11108 Watching for file changes with StatReloader
INFO 2023-09-21 09:23:02,360 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:04,204 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:05,179 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:05,901 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:08,516 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:11,795 basehttp 15416 10580 "GET /about/ HTTP/1.1" 200 2490
INFO 2023-09-21 09:23:58,508 autoreload 15416 11108 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:23:58,947 autoreload 14996 4340 Watching for file changes with StatReloader
INFO 2023-09-21 09:24:03,551 basehttp 14996 15064 "GET / HTTP/1.1" 200 2131
INFO 2023-09-21 09:24:05,482 basehttp 14996 15064 "GET /about/ HTTP/1.1" 200 2489
INFO 2023-09-21 09:24:49,534 autoreload 14996 4340 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:24:49,970 autoreload 12732 6452 Watching for file changes with StatReloader
INFO 2023-09-21 09:24:56,089 basehttp 12732 15936 "GET /about/ HTTP/1.1" 200 2477
INFO 2023-09-21 09:27:04,433 autoreload 12732 6452 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:27:04,865 autoreload 5164 4340 Watching for file changes with StatReloader
INFO 2023-09-21 09:27:11,268 basehttp 5164 13008 "GET /about/ HTTP/1.1" 200 2489
INFO 2023-09-21 09:37:41,254 autoreload 5164 4340 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:37:41,707 autoreload 4820 288 Watching for file changes with StatReloader
INFO 2023-09-21 09:41:21,808 autoreload 4820 288 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:41:22,324 autoreload 13368 3268 Watching for file changes with StatReloader
INFO 2023-09-21 09:41:27,484 basehttp 13368 12472 "GET / HTTP/1.1" 200 2131
INFO 2023-09-21 09:41:29,047 basehttp 13368 12472 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-09-21 09:41:29,073 log 13368 12472 Not Found: /about/img/me.jpg
WARNING 2023-09-21 09:41:29,074 basehttp 13368 12472 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-21 09:43:36,741 autoreload 13368 3268 D:\DjangoProject\homework_project\hw1_app\views.py changed, reloading.
INFO 2023-09-21 09:43:37,191 autoreload 16080 14688 Watching for file changes with StatReloader
INFO 2023-09-22 15:13:17,174 autoreload 2968 836 Watching for file changes with StatReloader
INFO 2023-09-22 15:13:24,229 basehttp 2968 9328 "GET / HTTP/1.1" 200 2131
WARNING 2023-09-22 15:13:24,376 log 2968 9328 Not Found: /favicon.ico
WARNING 2023-09-22 15:13:24,377 basehttp 2968 9328 "GET /favicon.ico HTTP/1.1" 404 2448
INFO 2023-09-22 15:13:26,811 basehttp 2968 9328 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-09-22 15:13:26,831 log 2968 9328 Not Found: /about/img/me.jpg
WARNING 2023-09-22 15:13:26,831 basehttp 2968 9328 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-09-22 15:13:28,339 basehttp 2968 9328 "GET / HTTP/1.1" 200 2131
INFO 2023-10-04 13:58:06,382 autoreload 10340 11672 Watching for file changes with StatReloader
INFO 2023-10-04 13:58:08,530 basehttp 10340 14784 "GET / HTTP/1.1" 200 2131
WARNING 2023-10-04 13:58:09,046 log 10340 14784 Not Found: /favicon.ico
WARNING 2023-10-04 13:58:09,046 basehttp 10340 14784 "GET /favicon.ico HTTP/1.1" 404 2448
INFO 2023-10-04 13:58:15,145 basehttp 10340 14784 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 13:58:15,162 log 10340 14784 Not Found: /about/img/me.jpg
WARNING 2023-10-04 13:58:15,162 basehttp 10340 14784 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 13:58:24,085 basehttp 10340 14784 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 13:58:24,096 log 10340 14784 Not Found: /about/img/me.jpg
WARNING 2023-10-04 13:58:24,096 basehttp 10340 14784 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 13:58:24,812 basehttp 10340 14784 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 13:58:24,825 log 10340 14784 Not Found: /about/img/me.jpg
WARNING 2023-10-04 13:58:24,826 basehttp 10340 14784 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 14:00:27,301 autoreload 15692 9328 Watching for file changes with StatReloader
INFO 2023-10-04 14:00:29,505 basehttp 15692 10656 "GET / HTTP/1.1" 200 2131
INFO 2023-10-04 14:00:31,232 basehttp 15692 10656 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 14:00:31,252 log 15692 10656 Not Found: /about/img/me.jpg
WARNING 2023-10-04 14:00:31,252 basehttp 15692 10656 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 14:00:36,391 basehttp 15692 10656 "GET / HTTP/1.1" 200 2131
INFO 2023-10-04 14:01:34,554 autoreload 7324 12628 Watching for file changes with StatReloader
INFO 2023-10-04 14:01:37,170 basehttp 7324 5488 "GET / HTTP/1.1" 200 2131
INFO 2023-10-04 14:01:39,070 basehttp 7324 5488 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 14:01:39,092 log 7324 5488 Not Found: /about/img/me.jpg
WARNING 2023-10-04 14:01:39,092 basehttp 7324 5488 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 14:13:52,979 autoreload 13680 3312 Watching for file changes with StatReloader
INFO 2023-10-04 14:14:05,167 basehttp 13680 8604 "GET / HTTP/1.1" 200 2131
INFO 2023-10-04 14:14:09,054 basehttp 13680 8604 "GET /about/ HTTP/1.1" 200 2373
WARNING 2023-10-04 14:14:09,078 log 13680 8604 Not Found: /about/img/me.jpg
WARNING 2023-10-04 14:14:09,079 basehttp 13680 8604 "GET /about/img/me.jpg HTTP/1.1" 404 2463
INFO 2023-10-04 15:27:57,406 autoreload 1556 15204 Watching for file changes with StatReloader
INFO 2023-10-09 13:03:50,000 autoreload 8648 12456 Watching for file changes with StatReloader
INFO 2023-10-09 13:06:04,010 autoreload 6920 2888 Watching for file changes with StatReloader
INFO 2023-10-09 13:08:09,366 autoreload 4084 5364 Watching for file changes with StatReloader
ERROR 2023-10-09 13:08:13,664 log 4084 6444 Internal Server Error: /
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'products' not found. 'products' is not a valid view function or pattern name.
ERROR 2023-10-09 13:08:13,668 basehttp 4084 6444 "GET / HTTP/1.1" 500 138564
WARNING 2023-10-09 13:08:13,830 log 4084 6444 Not Found: /favicon.ico
WARNING 2023-10-09 13:08:13,830 basehttp 4084 6444 "GET /favicon.ico HTTP/1.1" 404 2595
INFO 2023-10-09 13:09:05,616 autoreload 8324 11920 Watching for file changes with StatReloader
ERROR 2023-10-09 13:09:09,793 log 8324 2320 Internal Server Error: /
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'products' not found. 'products' is not a valid view function or pattern name.
ERROR 2023-10-09 13:09:09,796 basehttp 8324 2320 "GET / HTTP/1.1" 500 138779
INFO 2023-10-09 13:12:58,619 autoreload 10652 9040 Watching for file changes with StatReloader
ERROR 2023-10-09 13:13:03,209 log 10652 2660 Internal Server Error: /
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'products' not found. 'products' is not a valid view function or pattern name.
ERROR 2023-10-09 13:13:03,212 basehttp 10652 2660 "GET / HTTP/1.1" 500 138779
INFO 2023-10-09 13:14:51,383 autoreload 10652 9040 D:\DjangoProject\homework_project\hw2_app\views.py changed, reloading.
INFO 2023-10-09 13:14:51,888 autoreload 2140 9260 Watching for file changes with StatReloader
INFO 2023-10-09 13:16:00,167 autoreload 10468 6428 Watching for file changes with StatReloader
INFO 2023-10-09 13:16:02,360 basehttp 10468 13008 "GET / HTTP/1.1" 200 1395
ERROR 2023-10-09 13:17:14,828 log 10468 13008 Internal Server Error: /customer_orders/2/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:17:14,832 basehttp 10468 13008 "GET /customer_orders/2/ HTTP/1.1" 500 174342
INFO 2023-10-09 13:29:42,453 autoreload 8852 4140 Watching for file changes with StatReloader
ERROR 2023-10-09 13:29:44,865 log 8852 6428 Internal Server Error: /customer_orders/2/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 471, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:29:44,869 basehttp 8852 6428 "GET /customer_orders/2/ HTTP/1.1" 500 174471
INFO 2023-10-09 13:35:36,355 autoreload 9700 8324 Watching for file changes with StatReloader
INFO 2023-10-09 13:35:44,901 basehttp 9700 10500 "GET / HTTP/1.1" 200 1395
INFO 2023-10-09 13:36:02,356 basehttp 9700 10500 "GET /customer_orders/2 HTTP/1.1" 301 0
ERROR 2023-10-09 13:36:02,418 log 9700 9472 Internal Server Error: /customer_orders/2/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 715, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 847, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 914, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "D:\DjangoProject\homework_project\hw2_app\models.py", line 34, in get_absolute_url
    return reverse('order', kwargs={'pk': self.pk})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:36:02,426 basehttp 9700 9472 "GET /customer_orders/2/ HTTP/1.1" 500 191045
INFO 2023-10-09 13:42:50,839 autoreload 7116 11852 Watching for file changes with StatReloader
ERROR 2023-10-09 13:42:53,275 log 7116 13652 Internal Server Error: /customer_orders/2/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 715, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 847, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 914, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "D:\DjangoProject\homework_project\hw2_app\models.py", line 34, in get_absolute_url
    return reverse('order', kwargs={'pk': self.pk})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:42:53,281 basehttp 7116 13652 "GET /customer_orders/2/ HTTP/1.1" 500 191182
ERROR 2023-10-09 13:43:07,100 log 7116 13652 Internal Server Error: /customer_orders/3/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 715, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 847, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 914, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "D:\DjangoProject\homework_project\hw2_app\models.py", line 34, in get_absolute_url
    return reverse('order', kwargs={'pk': self.pk})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:43:07,105 basehttp 7116 13652 "GET /customer_orders/3/ HTTP/1.1" 500 192303
INFO 2023-10-09 13:43:55,197 autoreload 12468 11592 Watching for file changes with StatReloader
ERROR 2023-10-09 13:44:09,301 log 12468 11112 Internal Server Error: /customer_orders/2/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 175, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 157, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 167, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1005, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 238, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 966, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 1064, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 715, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 847, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 914, in _resolve_lookup
    current = current()
              ^^^^^^^^^
  File "D:\DjangoProject\homework_project\hw2_app\models.py", line 34, in get_absolute_url
    return reverse('order', kwargs={'pk': self.pk})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\urls\resolvers.py", line 828, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'order' not found. 'order' is not a valid view function or pattern name.
ERROR 2023-10-09 13:44:09,305 basehttp 12468 11112 "GET /customer_orders/2/ HTTP/1.1" 500 191045
INFO 2023-10-09 13:48:59,373 autoreload 12576 10528 Watching for file changes with StatReloader
INFO 2023-10-09 13:49:05,851 basehttp 12576 9908 "GET /customer_orders/2/ HTTP/1.1" 200 1703
INFO 2023-10-09 13:49:52,465 basehttp 12576 9908 "GET /customer_orders/1 HTTP/1.1" 301 0
INFO 2023-10-09 13:49:52,470 basehttp 12576 8892 "GET /customer_orders/1/ HTTP/1.1" 200 1809
INFO 2023-10-09 14:01:13,497 basehttp 12576 10516 "GET /customer_orders/1/ HTTP/1.1" 200 1809
INFO 2023-10-09 14:01:13,751 autoreload 12576 10528 D:\DjangoProject\homework_project\hw2_app\urls.py changed, reloading.
INFO 2023-10-09 14:01:14,235 autoreload 11048 10332 Watching for file changes with StatReloader
INFO 2023-10-09 14:01:17,054 basehttp 11048 10264 "GET /customer_orders/1/ HTTP/1.1" 200 1825
INFO 2023-10-09 14:01:21,383 basehttp 11048 10264 "GET /order/1/ HTTP/1.1" 200 123
INFO 2023-10-09 14:17:56,935 autoreload 11144 11644 Watching for file changes with StatReloader
INFO 2023-10-09 14:18:06,370 basehttp 11144 1544 "GET /customer_orders/1/ HTTP/1.1" 200 1825
ERROR 2023-10-09 14:18:08,214 log 11144 1544 Internal Server Error: /order/1/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 505, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'end'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 90, in rendered_content
    template = self.resolve_template(self.template_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\response.py", line 72, in resolve_template
    return select_template(template, using=self.using)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader.py", line 42, in select_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\backends\django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\engine.py", line 175, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\engine.py", line 157, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 200, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 293, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\loader_tags.py", line 232, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 513, in parse
    raise self.error(token, e)
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 511, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\defaulttags.py", line 853, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 507, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "D:\DjangoProject\venv\Lib\site-packages\django\template\base.py", line 558, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 14: 'end', expected 'empty' or 'endfor'. Did you forget to register or load this tag?
ERROR 2023-10-09 14:18:08,218 basehttp 11144 1544 "GET /order/1/ HTTP/1.1" 500 187861
INFO 2023-10-09 14:18:27,042 autoreload 12960 8296 Watching for file changes with StatReloader
INFO 2023-10-09 14:18:29,371 basehttp 12960 10516 "GET /order/1/ HTTP/1.1" 200 2032
INFO 2023-10-09 14:20:44,899 autoreload 3580 10428 Watching for file changes with StatReloader
INFO 2023-10-09 14:20:47,731 basehttp 3580 6216 "GET /order/1/ HTTP/1.1" 200 2079
INFO 2023-10-09 14:21:16,287 basehttp 3580 6216 "GET /order/1/ HTTP/1.1" 200 2079
INFO 2023-10-09 14:21:20,762 autoreload 13820 1172 Watching for file changes with StatReloader
INFO 2023-10-09 14:21:23,187 basehttp 13820 10052 "GET /order/1/ HTTP/1.1" 200 2081
INFO 2023-10-09 14:21:53,352 autoreload 10268 6244 Watching for file changes with StatReloader
INFO 2023-10-09 14:21:53,977 basehttp 10268 8788 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:21:56,400 basehttp 10268 8788 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:22:03,278 basehttp 10268 8788 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:22:13,558 autoreload 13276 8728 Watching for file changes with StatReloader
INFO 2023-10-09 14:22:17,228 basehttp 13276 12776 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:22:26,334 basehttp 13276 12776 "GET /order/1 HTTP/1.1" 301 0
INFO 2023-10-09 14:22:26,340 basehttp 13276 5948 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:22:36,309 basehttp 13276 5948 "GET /customer_orders/2/ HTTP/1.1" 200 1711
INFO 2023-10-09 14:22:39,241 basehttp 13276 5948 "GET /customer_orders/1/ HTTP/1.1" 200 1825
INFO 2023-10-09 14:22:41,178 basehttp 13276 5948 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:22:58,740 autoreload 11864 8804 Watching for file changes with StatReloader
INFO 2023-10-09 14:23:00,352 basehttp 11864 3196 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:23:02,328 basehttp 11864 3196 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:23:03,206 basehttp 11864 3196 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:23:11,087 basehttp 11864 3196 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:23:13,179 basehttp 11864 3196 "GET /customer_orders/1/ HTTP/1.1" 200 1825
INFO 2023-10-09 14:23:14,530 basehttp 11864 3196 "GET /order/3/ HTTP/1.1" 200 1949
INFO 2023-10-09 14:24:05,024 autoreload 2012 7732 Watching for file changes with StatReloader
INFO 2023-10-09 14:24:07,878 basehttp 2012 8624 "GET /order/3/ HTTP/1.1" 200 1949
INFO 2023-10-09 14:25:09,815 basehttp 2012 8624 "GET /customer_orders/1/ HTTP/1.1" 200 1825
INFO 2023-10-09 14:25:10,876 basehttp 2012 8624 "GET /order/1/ HTTP/1.1" 200 2083
INFO 2023-10-09 14:26:21,542 autoreload 14160 6040 Watching for file changes with StatReloader
INFO 2023-10-09 14:26:24,253 basehttp 14160 6372 "GET /order/1/ HTTP/1.1" 200 2084
INFO 2023-10-09 14:28:05,824 autoreload 8204 8804 Watching for file changes with StatReloader
INFO 2023-10-09 14:28:08,001 basehttp 8204 8512 "GET /order/1/ HTTP/1.1" 200 2040
INFO 2023-10-09 14:29:25,976 autoreload 5572 12672 Watching for file changes with StatReloader
INFO 2023-10-09 14:29:28,799 basehttp 5572 1820 "GET /order/1/ HTTP/1.1" 200 2062
INFO 2023-10-09 14:30:18,420 autoreload 13012 2040 Watching for file changes with StatReloader
INFO 2023-10-09 14:30:20,575 basehttp 13012 9296 "GET /order/1/ HTTP/1.1" 200 2079
INFO 2023-10-09 14:30:34,736 basehttp 13012 9296 "GET /customer_orders/1/ HTTP/1.1" 200 1825
INFO 2023-10-09 14:30:35,969 basehttp 13012 9296 "GET / HTTP/1.1" 200 1395
INFO 2023-10-09 14:31:03,005 basehttp 13012 9296 "GET /customer_orders/4 HTTP/1.1" 301 0
INFO 2023-10-09 14:31:03,011 basehttp 13012 5160 "GET /customer_orders/4/ HTTP/1.1" 200 1831
INFO 2023-10-09 14:59:18,344 autoreload 13760 7448 Watching for file changes with StatReloader
INFO 2023-10-09 15:00:58,373 autoreload 3068 14528 Watching for file changes with StatReloader
INFO 2023-10-09 15:01:17,114 basehttp 3068 14652 "GET /customer_products/2/7 HTTP/1.1" 301 0
ERROR 2023-10-09 15:01:17,179 log 3068 14148 Internal Server Error: /customer_products/2/7/
Traceback (most recent call last):
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\views\generic\base.py", line 104, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\views\generic\base.py", line 143, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\views\generic\base.py", line 226, in get
    context = self.get_context_data(**kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\homework_project\hw2_app\views.py", line 66, in get_context_data
    orders = Order.objects.filter(customer=customer,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\query.py", line 1436, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\query.py", line 1454, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\query.py", line 1461, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\sql\query.py", line 1545, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\sql\query.py", line 1576, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\sql\query.py", line 1426, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\sql\query.py", line 1236, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\DjangoProject\venv\Lib\site-packages\django\db\models\sql\query.py", line 1724, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'created_at' into field. Choices are: customer, customer_id, id, order_date, products, total_price
ERROR 2023-10-09 15:01:17,182 basehttp 3068 14148 "GET /customer_products/2/7/ HTTP/1.1" 500 132178
INFO 2023-10-09 15:03:09,539 autoreload 13660 9084 Watching for file changes with StatReloader
INFO 2023-10-09 15:03:12,570 basehttp 13660 13728 "GET /customer_products/2/7/ HTTP/1.1" 200 1757
INFO 2023-10-09 15:03:45,532 basehttp 13660 13728 "GET /customer_products/1/7/ HTTP/1.1" 200 1757
INFO 2023-10-09 15:04:06,585 basehttp 13660 13728 "GET /customer_products/1/8/ HTTP/1.1" 200 2150
INFO 2023-10-09 15:05:46,737 basehttp 13660 13728 "GET /customer_products/3/8/ HTTP/1.1" 200 2022
INFO 2023-10-09 15:06:30,190 basehttp 13660 13728 "GET /customer_products/4/8/ HTTP/1.1" 200 1898
INFO 2023-10-09 15:07:02,072 basehttp 13660 13728 "GET /customer_products/5/8/ HTTP/1.1" 200 2157
INFO 2023-10-09 15:07:16,812 basehttp 13660 13728 "GET /customer_products/6/8/ HTTP/1.1" 200 1765
WARNING 2023-10-09 15:07:25,204 log 13660 13728 Not Found: /customer_products/7/8/
WARNING 2023-10-09 15:07:25,205 basehttp 13660 13728 "GET /customer_products/7/8/ HTTP/1.1" 404 3216
INFO 2023-10-09 15:08:12,847 basehttp 13660 13728 "GET /customer_products/2/8 HTTP/1.1" 301 0
INFO 2023-10-09 15:08:12,852 basehttp 13660 9200 "GET /customer_products/2/8/ HTTP/1.1" 200 1895
INFO 2023-10-09 15:09:48,610 autoreload 13660 9084 D:\DjangoProject\homework_project\hw2_app\views.py changed, reloading.
INFO 2023-10-09 15:09:49,118 autoreload 4592 14024 Watching for file changes with StatReloader
INFO 2023-10-09 15:09:54,783 basehttp 4592 8992 "GET / HTTP/1.1" 200 1395
INFO 2023-10-09 15:14:29,273 autoreload 4592 14024 D:\DjangoProject\homework_project\hw2_app\views.py changed, reloading.
INFO 2023-10-09 15:14:29,800 autoreload 740 532 Watching for file changes with StatReloader
INFO 2023-10-09 15:23:19,221 autoreload 14500 7040 Watching for file changes with StatReloader
INFO 2023-10-09 15:23:28,409 basehttp 14500 14876 "GET /customer_products/2/8/ HTTP/1.1" 200 1895
INFO 2023-10-09 15:23:36,239 basehttp 14500 14876 "GET /customer_products/2/7/ HTTP/1.1" 200 1895
INFO 2023-10-09 15:44:28,049 autoreload 14500 7040 D:\DjangoProject\homework_project\hw2_app\views.py changed, reloading.
INFO 2023-10-09 15:44:28,508 autoreload 16348 14784 Watching for file changes with StatReloader
